server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndexer.java

```diff
@@ -90,19 +85,22 @@ private static long doIndex(BulkIndexer bulk, Iterator<ActiveRuleDoc> activeRule
   }
 
   public void index(List<ActiveRuleChange> changes) {
-    deleteKeys(FluentIterable.from(changes)
-      .filter(MatchDeactivatedRule.INSTANCE)
-      .transform(ActiveRuleChangeToKey.INSTANCE)
-      .toList());
+    deleteKeys(changes.stream()
+      .filter(c -> c.getType().equals(ActiveRuleChange.Type.DEACTIVATED))
+      .map(c -> c.getKey())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1612)